app/models.py

```diff
@@ -722,7 +733,7 @@ class ServiceTableMixin(object):
     # Service publishing time.
     service_id = db.Column(db.String, index=True, unique=True, nullable=False)
 
-    data = db.Column(JSON)
+    data = db.Column(JSON, nullable=False, default=dict)
```

Comment:
This just felt like the less risky change (not knowing all the possible ways to create that row). Happy to change if you're confident that an empty dict is an error just as much as NULL would be.